lntest/utils/wait.go

```diff
@@ -0,0 +1,78 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+)
+
+// WaitPredicate is a helper test function that will wait for a timeout period
```

Comment:
I don't think we really need a utils package here. What's the downside of them just living in the main `lntest` package? Or is this an attempt to resolve an import cycle/ 

The module import path is also much clearer as `lntest` vs `utils`. `utils` as a package name is ambiguous (what type of utils, just tests, everything?). In general `utils` is a bad package name, as there's no clear stance of exactly _what_ goes into the package.

Reply:
If it's an import cycle thing, then there's no harm in just copying over the exact function that's needed elsewhere to address the import cycle. Each of these functions are no more than 20 lines long, so there's no extreme duplication or multiplication of maintenance costs by copying the function needed.

Reply:
yes there's an import cycle because `lntest` depends on `routing`. the functions were originally copied here, but decided it best to resolve the import cycle now so that we don't end up an unbounded number copies as we continue to use them throughout the code base.

> In general utils is a bad package name, as there's no clear stance of exactly what goes into the package.

The package is `lntest/utils`, not just `utils`. idt it's unclear what kind of utils would go there given the scoping. If an import collision arrises it's trivial to rename it within the scope of the file. 

`testutils` is imo too long for something used so prevalently and is ideally an invisible package; it's also redundant in the scope of `lntest/testutils`. i don't really have a better suggestion, but i think _some_ package should be created here

Reply:
perhaps a package called `wait`?

Reply:
Ah I see, since it refs `routing.BlockPadding`.... 


>  idt it's unclear what kind of utils would go there given the scoping

I'd agree if devs were forced to type out `lntest/util.Func` each time, but instead one sees the `utils` package being referenced and is left to wonder exactly what else that package provides or where it's being import from until they scroll back up to the main import list. 

> perhaps a package called `wait`?

I'm cool with `wait`.

Reply:
`lntest/wait` sgtm

Reply:
My opinion about this:
* I don't like copy/paste
* A package for wait functions will prompt a rename when a non-wait shared test function is added. Or another package.
* I can live with `testutils` as a package name, even though a bit long

How about putting the generic wait utils in `lntest` and moving things that are currently in `lntest` to a more descriptive package like `lntest/harness`?

Btw, I don't see the dependency of `lntest` on `routing`, just that `lntest/itest` depends on `routing`.

Reply:
okay, went with `lntest/wait`, can always change it later if we have better ideas.

> Btw, I don't see the dependency of lntest on routing, just that lntest/itest depends on routing.

`lntest` depends on `routerrpc` which depends on `routing`

> A package for wait functions will prompt a rename when a non-wait shared test function is added. Or another package.

I think this is what we want (adding a new package), keeping them small will keep the deps small and minimize import cycles going forward.

Reply:
> I'd agree if devs were forced to type out lntest/util.Func each time, but instead one sees the utils package being referenced and is left to wonder exactly what else that package provides or where it's being import from until they scroll back up to the main import list.

I presume devs would only have to do this once, and then remember what the package is. for me scrolling to the top of the file once is an acceptable tradeoff